src/main/java/io/fabric8/maven/docker/model/ContainerDetails.java

```diff
@@ -121,6 +126,21 @@ public boolean isRunning() {
         return state.getBoolean(RUNNING);
     }
 
+    @Override
+    public boolean isHealthy() {
+        final JSONObject state = json.getJSONObject(STATE);
+        // always indicate healthy for docker hosts that do not support health checks.
+        return !state.has(HEALTH) || HEALTH_STATUS_HEALTHY.equals(state.getJSONObject(HEALTH).getString(STATUS));
+    }
+
+    @Override
+    public String healthcheckTests() {
+        if (!json.getJSONObject(CONFIG).has(HEALTHCHECK)) {
+            return null;
+        }
+        return json.getJSONObject(CONFIG).getJSONObject(HEALTHCHECK).getJSONArray(TEST).join(", ");
```

Comment:
I just saw we could add this check simply to a `WatchConfiguration.initAndValidate()` much like we do for `BuildImageConfiguration.initAndValidate()`. This method is supposed to return [a minimal API version](https://github.com/rhuss/docker-maven-plugin/blob/d431b076a59d97217e2df58ff9d2771a0d83d82e/src/main/java/io/fabric8/maven/docker/config/ConfigHelper.java#L74-L74) to operate. This would result to refuse to work when a [too old Docker daemon is used](https://github.com/rhuss/docker-maven-plugin/blob/7f5d0edd2c29edebb28493313df320a9a3807968/src/main/java/io/fabric8/maven/docker/service/DockerAccessFactory.java#L46-L46).

The implementation should be simple and straight forward.

BTW, I think HEALTHCHECK is supported since 1.24.